lib/phase/flatten.js

```diff
@@ -307,6 +307,46 @@ function flattenSharedIntoJobs(shared, jobs) {
     return newJobs;
 }
 
+/**
+ * Merge shared steps into job when mergeSharedSteps annotation is set to true
+ * @param  {Object} sharedConfig               Shared config
+ * @param  {Object} jobConfig                  Job config
+ * @param  {Object} [template]                 Template
+ * @return {Object}                            Merged job config
+ */
+function handleMergeSharedStepsAnnotation({ sharedConfig, jobConfig, template }) {
+    const sharedSteps = sharedConfig.steps;
+    const newJobConfig = jobConfig;
+
+    // merge shared steps into oldJob
+    if (
+        sharedSteps &&
+        ((jobConfig.annotations && jobConfig.annotations['screwdriver.cd/mergeSharedSteps']) ||
+            ((jobConfig.annotations === undefined ||
+                jobConfig.annotations['screwdriver.cd/mergeSharedSteps'] === undefined) &&
+                template &&
+                template.annotations &&
+                template.annotations['screwdriver.cd/mergeSharedSteps']))
+    ) {
```

Comment:
if a job is not using a template, will it go into this condition? 
Does the second example in [this merging-shared-steps document](https://docs.screwdriver.cd/user-guide/templates/job-templates#merging-with-shared-steps) even valid?

Reply:
no it's missing a template we talked about this